tests/integration/test_events.py

```diff
@@ -152,6 +154,96 @@ def get_message(queue_url):
         # clean up
         self.cleanup(bus_name, rule_name, target_id, queue_url=queue_url)
 
+    def test_put_events_with_target_sqs_event_detail_match(self):
+        queue_name = 'queue-{}'.format(short_uid())
+        rule_name = 'rule-{}'.format(short_uid())
+        target_id = 'target-{}'.format(short_uid())
+        bus_name = 'bus-{}'.format(short_uid())
+
+        sqs_client = aws_stack.connect_to_service('sqs')
+        queue_url = sqs_client.create_queue(QueueName=queue_name)['QueueUrl']
+        queue_arn = aws_stack.sqs_queue_arn(queue_name)
+
+        self.events_client.create_event_bus(
+            Name=bus_name
+        )
+
+        self.events_client.put_rule(
+            Name=rule_name,
+            EventBusName=bus_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN_1)
+        )
+
+        rs = self.events_client.put_targets(
+            Rule=rule_name,
+            EventBusName=bus_name,
+            Targets=[
+                {
+                    'Id': target_id,
+                    'Arn': queue_arn,
+                    'InputPath': '$.detail'
+                }
+            ]
+        )
+
+        self.assertIn('FailedEntryCount', rs)
+        self.assertIn('FailedEntries', rs)
+        self.assertEqual(rs['FailedEntryCount'], 0)
+        self.assertEqual(rs['FailedEntries'], [])
+
+        self.events_client.put_events(
+            Entries=[{
+                'EventBusName': bus_name,
+                'Source': TEST_EVENT_PATTERN['Source'][0],
+                'DetailType': TEST_EVENT_PATTERN['detail-type'][0],
+                'Detail': json.dumps({'EventType': '1'})
+            }]
+        )
+
+        def get_message(queue_url):
+            resp = sqs_client.receive_message(QueueUrl=queue_url)
+            return resp.get('Messages')
+
+        messages = retry(get_message, retries=3, sleep=1, queue_url=queue_url)
+        self.assertEqual(len(messages), 1)
+
+        actual_event = json.loads(messages[0]['Body'])
+        self.assertEqual(actual_event, {'EventType': '1'})
+
+        self.events_client.put_events(
+            Entries=[{
+                'EventBusName': bus_name,
+                'Source': TEST_EVENT_PATTERN['Source'][0],
+                'DetailType': TEST_EVENT_PATTERN['detail-type'][0],
+                'Detail': json.dumps({'EventType': '2'})
+            }]
+        )
+
+        def get_message(queue_url):
+            resp = sqs_client.receive_message(QueueUrl=queue_url)
+            return resp.get('Messages', [])
+
+        messages = retry(get_message, retries=3, sleep=1, queue_url=queue_url)
+        self.assertEqual(len(messages), 0)
+
+        # clean up
+        sqs_client.delete_queue(QueueUrl=queue_url)
+
+        self.events_client.remove_targets(
```

Comment:
Can we use (or extend, if necessary) the existing `self.cleanup(...)` method here?